libpod/oci_conmon_common.go

```diff
@@ -1311,38 +1305,17 @@ func (r *ConmonOCIRuntime) createOCIContainer(ctr *Container, restoreOptions *Co
 }
 
 // configureConmonEnv gets the environment values to add to conmon's exec struct
-// TODO this may want to be less hardcoded/more configurable in the future
-func (r *ConmonOCIRuntime) configureConmonEnv(runtimeDir string) []string {
-	var env []string
-	for _, e := range os.Environ() {
-		if strings.HasPrefix(e, "LC_") {
-			env = append(env, e)
-		}
-		if strings.HasPrefix(e, "LANG=") {
-			env = append(env, e)
+func (r *ConmonOCIRuntime) configureConmonEnv() []string {
+	env := os.Environ()
+	res := make([]string, 0, len(env))
+	for _, v := range env {
+		if strings.HasPrefix(v, "NOTIFY_SOCKET=") {
+			// The NOTIFY_SOCKET cannot leak into the environment.
+			continue
```

Comment:
1) I think you can remove the startCommand() function workaround with this change

2) Looking at the test failure I think we also need to unset XDG_RUNTIME_DIR when run as root, the problem is that when you login with ssh for example we also get `XDG_RUNTIME_DIR=/run/user/0` as root so this makes crun store its files under a different location thus failing to find it.